crypto/x509/v3_alt.c

```diff
@@ -121,6 +121,17 @@ STACK_OF(CONF_VALUE) *i2v_GENERAL_NAME(X509V3_EXT_METHOD *method,
                 return NULL;
             break;
         default:
+             /* check if the value is something printable */
+            if (gen->d.otherName->value->type == V_ASN1_IA5STRING)
+                 if (X509V3_add_value("othername",
```

Comment:
The Travis failure is relevant. Please use X509V3_add_value_uchar(). Also you're losing the information about the type_id - perhaps using OBJ_obj2txt() to add it would make sense?

Reply:
understood. fixing. 

Not using the suggested OBJ_OBJ_obj2txt() as these particular strings are reall that- textual strings which we want to regex/match on; for example in an Apache/webserver TLS lockdown or when doing a RFC 3161 signature. And you'd end up having to match on a <ascii-ordinal>.<ascii-ordinal>.<32><2E> .. etc 

The latter which is not really openSSL its fault - the design behind these othernames is abit silly (encoding OIDs as ascii strings that is and then ascii-concatenating them with a dash)

Reply:
But I am talking about the type_id not about the value itself. I do not understand - is the information in type_id duplicated in the value itself?

Reply:
In this case it is devoid of info; in the case of the customs document it is just '.1/ISO'. Let me check a few others to see if there is a useful value sometimes.

Reply:
Ok - so I found some other cases where it is useful. Below would work generic enough; and with a bit of a fiddle with snprintf to leave the `othername:` in place.

```
@@ -79,6 +79,7 @@ STACK_OF(CONF_VALUE) *i2v_GENERAL_NAME(X509V3_EXT_METHOD *method,
                                        STACK_OF(CONF_VALUE) *ret)
 {
     unsigned char *p;
+    char othername[256];
     char oline[256], htmp[5];
     int i;
 
@@ -121,22 +122,27 @@ STACK_OF(CONF_VALUE) *i2v_GENERAL_NAME(X509V3_EXT_METHOD *method,
                 return NULL;
             break;
         default:
+             if (OBJ_obj2txt(oline, sizeof(oline), gen->d.otherName->type_id, 0)) 
+                 snprintf(othername, sizeof(othername), "othername: %s:", oline);
+             else
+                 strncpy(othername, "othername", sizeof(othername));
+
              /* check if the value is something printable */
             if (gen->d.otherName->value->type == V_ASN1_IA5STRING) {
-                 if (X509V3_add_value_uchar("othername",
+                 if (X509V3_add_value_uchar(othername,
                              gen->d.otherName->value->value.ia5string->data,
                              &ret)) {
                      return ret;
                 };
             };
             if (gen->d.otherName->value->type == V_ASN1_UTF8STRING) {
-                if (X509V3_add_value_uchar("othername",
+                if (X509V3_add_value_uchar(othername,
                              gen->d.otherName->value->value.utf8string->data,
                              &ret)) {
                      return ret;
                 };
             };
-            if (!X509V3_add_value("othername", "<unsupported>", &ret))
+            if (!X509V3_add_value(othername, "<unsupported>", &ret))
                 return NULL;
             break;
         }
```

as that would pretty much remove the need for all of them (as those OIDs nicely resolve through OBJ_obj2txt).
And this preserves the double :: of the NAIRealm, UNP and others above it.

But I am wondering if it makes sense to also make the same change to all those cases

```
  case NID_ms_upn:
            if (gen->d.otherName->value->type != V_ASN1_UTF8STRING
                    || !X509V3_add_value_uchar("othername: UPN:",
                            gen->d.otherName->value->value.utf8string->data,
                            &ret))
                return NULL;
            break;
   
```

Is that in line with what you where thinkng ?

Reply:
It would just remove the need for that whole case statement ?

Reply:
It would change the output for the UPN and SmtpUTF8Mailbox so these would have to be probably special-cased. However yes, it could be simplified like this.

Reply:
fair point - and I found one Linux "API" in the /etc/cloud which then breaks. So lets not do this.